src/openrct2/actions/PlacePeepSpawnAction.hpp

```diff
@@ -96,27 +97,26 @@ DEFINE_GAME_ACTION(PlacePeepSpawnAction, GAME_COMMAND_PLACE_PEEP_SPAWN, GameActi
     {
         auto res = std::make_unique<GameActionResult>();
         res->Expenditure = ExpenditureType::LandPurchase;
+        // TODO: Suspicious z/8 to transform Coords into Coords
         res->Position = CoordsXYZ{ _location.x, _location.y, _location.z / 8 };
 
         // If we have reached our max peep spawns, remove the oldest spawns
         while (gPeepSpawns.size() >= MAX_PEEP_SPAWNS)
         {
             auto oldestSpawn = gPeepSpawns.begin();
-            auto oldX = oldestSpawn->x;
-            auto oldY = oldestSpawn->y;
+            auto oldSpawnPos = CoordsXY{ oldestSpawn->x, oldestSpawn->y };
             gPeepSpawns.erase(oldestSpawn);
-            map_invalidate_tile_full({ oldX, oldY });
+            map_invalidate_tile_full(oldSpawnPos);
         }
 
         // Shift the spawn point to the middle of the tile
-        int32_t middleX, middleY;
-        middleX = _location.x + 16 + (word_981D6C[_location.direction].x * 15);
-        middleY = _location.y + 16 + (word_981D6C[_location.direction].y * 15);
+        auto middlePos = CoordsXY{ _location } + CoordsXY{ 16, 16 }
+            + CoordsXY{ word_981D6C[_location.direction].x * 15, word_981D6C[_location.direction].y * 15 };
```

Comment:
This isn't actually shifting it to the middle of the tile. I suspect what this is meant to do is call `ToTileCentre()` then add on  `word_981D6C` which sets the actual spawn point to be the very edge of the tile. (each tile is 32, 32 spawn will be at 16 + 15 = 31)

Reply:
Can you therefore change it to reflect that. i.e. reword comment and change to `auto spawnPos = _location.ToTileCentre() + CoordsXY{ word_981D6C[_location.direction].x * 15, word_981D6C[_location.direction].y * 15 };`

Reply:
Alright, done